### PR TITLE
For #42: test for GoSlice

### PR DESCRIPTION
**Pull request by olenagerasimova**
Created test for `GoSlice` for #42.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#43](https://github.com/artipie/go-adapter/issues/43) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/f242677a-cb9e-4923-9d04-c811b18500a7); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/http/GoSliceTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    void returnsInfo() throws ExecutionException, InterruptedException {
+        final Storage storage = new InMemoryStorage();
+        final String path = "news.info/some/day/@v/v0.1.info";
+        final String body = "{\"Version\":\"0.1\",\"Time\":\"2020-01-24T00:54:14Z\"}";
+        storage.save(
+            new KeyFromPath(path),
+            new Content.From(body.getBytes())
+        ).get();
+        MatcherAssert.assertThat(
+            new GoSlice(storage).response(
```
Comment (Vatavuk) [inline src/test/java/com/artipie/http/GoSliceTest.java:62]: @olenagerasimova we can use `GoSliceTest.get(path, body)` as in other tests

File: src/test/java/com/artipie/http/GoSliceTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import org.hamcrest.core.AllOf;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for {@link GoSlice}.
+ * @since 0.3
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class GoSliceTest {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/http/GoSliceTest.java:50]: @olenagerasimova I think we need to add a test for `latest`

File: src/test/java/com/artipie/http/GoSliceTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    /**
+     * Composes storage.
+     * @param path Where to store
+     * @param body Body to store
+     * @return Storage
+     * @throws ExecutionException On error
+     * @throws InterruptedException On error
+     */
+    private static Storage get(final String path, final String body)
```
Comment (Vatavuk) [inline src/test/java/com/artipie/http/GoSliceTest.java:156]: @olenagerasimova I suggest that we rename this method to `storage` to be more obvious in tests what this method do.

**Comment by 0crat** [comment]
Job was finished in 21 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**